### PR TITLE
Support for Markdown decoding

### LINKED ISSUE #46: Support for markdown decoding
Hi,

We need a wysiwyg markdown editor in our app and we are considering using Fleather package.
For this we need to initialize the editor with pre-existing markdown data and also extract markdown from the editor data.
We only need basic markdown for now.

I've spotted the `ParchmentMarkdownCodec.encode` function that let us convert deltas to markdown formatted text, but the `ParchmentMarkdownCodec.decode` isn't implemented yet and if I'm right it's needed to load markdown formatted text inside the editor.

Is there any way to do this actually?

### PR DESCRIPTION
**Pull request by amantoux**
Closes #46 

Many thanks to @nalbion for the initial PR

### CODE REVIEW

**Comment by amantoux** [comment]
Hey @Amir-P can you please take a look?

File: packages/parchment/lib/src/convert/markdown.dart [hunk 1]
```diff
@@ -210,8 +495,18 @@ class _ParchmentMarkdownEncoder extends Converter<Delta, String> {
...
 
-      final tag = kSimpleBlocks[block];
+      final tag = simpleBlocks[block];
       buffer.write(tag);
     }
   }
 }
+
+extension<E> on Iterable<E> {
+  Iterable<E> reverse() {
```
Comment (Amir-P) [inline packages/parchment/lib/src/convert/markdown.dart:505]: I've tested this functions performance (I suspected it's not going to perform well) and I found that it's way faster to do `iterable.toList().reversed` than to do `iterable.reverse()`. Actually it's 1350 times faster!

**Review by Amir-P — CHANGES_REQUESTED** [review]
LGTM although I didn't read decoder line by line. I just have a few small change requests.

File: packages/parchment/test/convert/markdown_test.dart [hunk 2]
```diff
@@ -10,10 +10,360 @@ import 'package:test/test.dart';
...
+      expect(delta.elementAt(8).data, 'text in italics');
+      expect(delta.elementAt(8).attributes?['block'], 'quote');
+      expect(delta.elementAt(8).attributes?['i'], true);
+      expect(delta.elementAt(8).attributes?.length, 2);
+
+      final andBack = parchmentMarkdown.encode(delta);
+      expect(andBack, markdown);
+    });
+
+    /*test('nested bq', () {
```
[the remark this answers is not in the document]
Comment (amantoux) [reply packages/parchment/test/convert/markdown_test.dart:347]: Yes we'll ignore

**Review by Amir-P — APPROVED** [review]
LGTM.

**Pull request merged**